unittest/tfreedv_mode_800XA.c

```diff
@@ -0,0 +1,133 @@
+
+  FILE........: tfreedv_mode_800XA.c
+  AUTHOR......: Jeroen Vreeken
+  DATE CREATED: 24 May 2020
+
+  Tests specific for mode 800XA
+
+\*---------------------------------------------------------------------------*/
+/*
+  Copyright (C) 2020 Jeroen Vreeken <jeroen@vreeken.net>
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with this program; if not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdio.h>
+#include "freedv_api.h"
+
+int main(int argc, char **argv)
+{
+    struct freedv *f;
+    int i;
+
+    printf("freedv_api tests for mode 800XA\n");
+
+    printf("freedv_open(FREEDV_MODE_800XA) ");
+    f = freedv_open(FREEDV_MODE_800XA);
+    if (!f) {
+        printf("Failed to open\n");
+        goto fail;
+    }
+    printf("Passed\n");
+
+    printf("freedv_get_mode() ");
+    int mode = freedv_get_mode(f);
+    if (mode != FREEDV_MODE_800XA) {
+        printf("mode %d does not match FREEDV_MODE_800XA %d\n", mode, FREEDV_MODE_800XA);
+        goto fail;
+    }
+    printf("Passed\n");
+
+    printf("freedv_get_n_max_modem_samples() ");
+    int max_samples = freedv_get_n_max_modem_samples(f);
+    if (max_samples != 660) {
+        printf("modem max samples %d != 660\n", max_samples);
+        goto fail;
+    }
+    printf("%d Passed\n", max_samples);
+
+    printf("freedv_get_n_nom_modem_samples() ");
+    int nom_samples = freedv_get_n_nom_modem_samples(f);
+    if (nom_samples != 640) {
+        printf("modem nom samples %d != 640\n", nom_samples);
+        goto fail;
+    }
+    printf("%d Passed\n", nom_samples);
+
+    printf("freedv_get_n_speech_samples() ");
+    int speech_samples = freedv_get_n_speech_samples(f);
+    if (speech_samples != 640) {
+        printf("Expected 640 speech samples, got %d\n", speech_samples);
+        goto fail;
+    }
+    printf("%d Passed\n", speech_samples);
+
+    printf("freedv_get_n_bits_per_codec_frame() ");
+    int codec_bits = freedv_get_bits_per_codec_frame(f);
+    if (codec_bits != 28) {
+        printf("Expected 28 codec bits, got %d\n", codec_bits);
+	goto fail;
+    }
+    printf("%d Passed\n", codec_bits);
+
+    printf("freedv_get_n_bits_per_modem_frame() ");
+    int frame_bits = freedv_get_bits_per_modem_frame(f);
+    if (frame_bits != 56) {
+        printf("Expected 56 codec bits, got %d\n", frame_bits);
+	goto fail;
+    }
+    printf("%d Passed\n", frame_bits);
+
```

Comment:
@JeroenVreeken these tests are testing constants, is there any conceivable way they could fail? I am wondering if they do anything useful. Also you could just do them in one line with asserts:
```
assert(freedv_get_bits_per_modem_frame(f) == 56);
```

Reply:
Some of them might be a bit pedantic, but others like bits per modem frame are not pure constants, they are a calculated based on the bits per codec frame.
One thing I learned with unittests over the last few years is to especially test the stupid stuff. Its the first that gets broken :)

Reply:
Yes, good point, some are computed so that makes sense.